packages/belos/doc/parameterList/createValidParameterList.cpp

```diff
@@ -86,6 +92,38 @@ void writeXSLHeader( std::ofstream& out )
           << "href=\"common/parameterList/parameterList.xsl\"?>\n";
 }
 
+// This test is a special case with no libs to provide the new DII registration.
+// SolverFactoryParent is now going to run a specialized form of
+// registerSolverFactoryForLib based on the template parameters. For this test
+// that will pick up Tpetra and attempt to call the following method:
+// Belos::Details::Tpetra::registerSolverFactory()
+// For this particular test that won't be available so I have manually recreated
+// it here and then register just the managers actually used in this test.
+// TODO: Decide how we want to handle this  in the new DII setup.
```

Comment:
Currently this is only handling 4 tpetra cases but I can see it is eventually intended to handle more. I changed this back to its original form and then simply added the cpp for registering tpetra managers to the CMakeLists. Usually a factory which builds tpetra managers would link to the belostpetra.lib to call registerSolverFactory() but this special case doesn't have the libs. Providing the cpp for tpetra registerSolverFactory() in the CMakeLists.txt is sufficient to make this work.

That does mean this test is rebuilding the entire tpetra set. It would be more efficient if things could be organized so this links to the libs and avoid the duplication. I haven't looked at the details of this carefully yet so I'm not sure if that would be ruled out for some reason. Should I change this to be linked to those libs?

Then it sounds like we would want the belos and epetra managers registered as well.

Note this is not currently using different MV and OP types. It is just the standard tpetra forms, though other tests commented on below have special forms so I added a registerCustomSolverFactory to handle them.